### PR TITLE
[#5] update to RxSwift 6, remove ReactiveCall

### PR DESCRIPTION
**Pull request by rm98**
* update to RxSwift 6 (migration: renaming .error to .failure, remove ReactiveCall)
* add optimistic version matching to Podfile and RxSession.podspec
* add Gemfile
* add GitHub Action to execut pod lib lint on push

### CODE REVIEW

File: RxSession/RxSession.swift [hunk 1]
```diff
@@ -55,26 +55,3 @@ extension Session: ReactiveSession {
         }
     }
 }
-
-/// Convenience Protocol to support call.rx
-public protocol ReactiveCall: Call, ReactiveCompatible {
-    
-}
-
-extension Reactive where Base: ReactiveCall {
```
Comment (mariohahn) [inline RxSession/RxSession.swift:64]: Warum entfernen wir das?

**Review by oanhof — APPROVED** [review]

**Review by mariohahn — APPROVED** [review]

**Comment by oanhof** [comment]
Should we add the helper functions in a different way than the `ReactiveCompatible` extension? I think there should at least be a short readme to explain how to use this lib.

File: RxSession/RxSession.swift (hunk 1 above)
Comment (rm98) [reply RxSession/RxSession.swift:64]: Weil `ReactiveCompatible` jetzt `: AnyObject` ist, die Calls mit `Call` und `ReactiveCall` aber eigentlich meist `struct` verwenden.

**Comment by rm98** [comment]
> Should we add the helper functions in a different way than the `ReactiveCompatible` extension? I think there should at least be a short readme to explain how to use this lib.

Do you think this is necessary @oanhof?

I don't think the use case of directly starting calls is that prevalent, and this extension until now:
* prevented enabling logging of the `Session`
* prevented choosing the used `URLSession`

**Comment by oanhof** [comment]
> > Should we add the helper functions in a different way than the `ReactiveCompatible` extension? I think there should at least be a short readme to explain how to use this lib.
> 
> Do you think this is necessary @oanhof?
> 
> I don't think the use case of directly starting calls is that prevalent, and this extension until now:
> 
> * prevented enabling logging of the `Session`
> * prevented choosing the used `URLSession`

I agree with you. I think I never actually used these functions in any project. I'm ok with removing them.

**Comment by rm98** [comment]
@oanhof I've added a short `README.md` and a `CHANGELOG.md`

**Pull request merged**